pkg/planner/core/memtable_infoschema_extractor.go

```diff
@@ -150,7 +159,8 @@ func (e *InfoSchemaBaseExtractor) ListSchemasAndTables(
 			findTablesByID(is, tableIDs, tableNames, tableMap)
 			tableSlice := maps.Values(tableMap)
 			tableSlice = filterSchemaObjectByRegexp(e, ec.table, tableSlice, extractStrTableInfo)
-			return findSchemasForTables(ctx, is, schemas, tableSlice)
+			schemas, unspecified := e.listExtractedSchemas(is)
```

Comment:
By the way, it's a bit weird to check predicates by `ec.tableID != ""`

Reply:
Do you mean we should decide this at the compile time? Do you have any suggestions?

Reply:
Nevermind, because I have no suggestion yet :(